### ISSUE TITLE
Production error using Rx2Firebase

### ISSUE THREAD

**Issue by mradzinski** [FrangSierra/Rx2Firebase#37]
Hi @FrangSierra, I have plenty of users hitting non-fatal exceptions pointing to:

```
Non-fatal Exception: durdinapps.rxfirebase2.exceptions.RxFirebaseDataException
       at durdinapps.rxfirebase2.RxFirebaseDatabase$3$1.onComplete(RxFirebaseDatabase.java:130)
       at com.google.android.gms.internal.zzeck.run(Unknown Source)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:7007)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```
As per the decompiled class, the error happens on `public static Completable setValue(@NonNull final DatabaseReference ref, final Object value)`, but that method doesn't seems to be using any `RxFirebaseDataException` instance. I can't seem to find any logic nor explanation to why this might be happening 😕 . I'm a little bit puzzled with this, so any help you might be able to provide will be awesome!

Thanks man!

**Comment by FrangSierra** [comment]
Umm, the implementation as you say is quite simple, I can't understand what is happening in this case:

```
 @NonNull
   public static Completable setValue(@NonNull final DatabaseReference ref,
                                      final Object value) {
      return Completable.create(new CompletableOnSubscribe() {
         @Override
         public void subscribe(@NonNull final CompletableEmitter e) throws Exception {
            ref.setValue(value).addOnSuccessListener(new OnSuccessListener<Void>() {
               @Override public void onSuccess(Void aVoid) {
                  e.onComplete();
               }
            }).addOnFailureListener(new OnFailureListener() {
               @Override public void onFailure(@NonNull Exception exception) {
                  e.onError(exception);
               }
            });
         }
      });
   }
```
Could you provide me the subcription in your code? also print the non-fatal error in your `onError` call in the subscription.

If it still happening, copy the method from the library and do a test, switching `.addOnSucessListener` and `addOnFailureListener` with a single `addOnCompleteListener` and checking with an ´if´ if the task have an error, emitting the right value(you can check other implementations from the library with this model). If doing this your error stops, please let me know and I will switch it in the code.

Thank you for your support and use the library!

**Comment by mradzinski** [comment]
Hey @FrangSierra I can't actually know which subscription is responsible for this, there're over 200 subscriptions. What I can tell you is that each and every single one of them logs the error in an appropriate way so I'm guessing for some really odd reason this isn't even reaching the subscription's `onError`.

I'll try replacing the method with the implementation you mention, but I doubt it'll be of help given that `addOnCompleteListener` is pretty much the same as `addOnSucessListener` and `addOnFailureListener`.

Let you know in a day or two if it keeps happening once I release a new version.

**Comment by mradzinski** [comment]
Hey @FrangSierra, I actually tried switching `addOnSucessListener` and `addOnFailureListener` with a single `addOnCompleteListener` and actually it got much worse (duplicated the amount of non-fatals). I went through all subscriptions in the app and double checked errors where being handled, so I'm mostly sure `onError` is not even being invoked for this exception.

Any ideas why `onError` can fail to be invoked?

**Comment by FrangSierra** [comment]
I'm running tests to check if the onError is call and also trying to force and error using the method on my local machine and everything is working propperly. Its quite weird that it is failing all all the calls in your app. I will continue researching to find the bug and I will try to find a solution

**Comment by mradzinski** [comment]
@FrangSierra I've been researching this as well. I might have found the reason for this to be happening. It might be related to a bug in our app which is manifesting wrongly through your library. Let me publish a new version with that fixed (It'll be around Friday this week) and I let you know of the outcome.

**Comment by FrangSierra** [comment]
Thank you @mradzinski . If you need some help or I can help you in something, dont doubt in contact me!

**Comment by FrangSierra** [comment]
Hi @mradzinski ,

Any new about this?

**Comment by FrangSierra** [comment]
I added a few checks on the `onFailure` of the library that maybe was causing this error on your calls, anyway I have done a lot of tests using the methods and I don't understand why this can be happening. At the moment it doesnt look to be a problem of the library, so please, keep me updated about this topic. 

I will close the issue, feel free to re-open it.

**Comment by mradzinski** [comment]
Hey @FrangSierra We just released yesterday, so I took some time and waited before calling it a victory. It seems my original assumption that the issue was related to a bug in our app which was manifesting wrongly through your library was right. As soon as we fixed the bug the non-fatal errors stopped showing up. So yeah, lets keep this issue closed and locked.

Thanks for your assistance and time!!! I owe you a couple beers!
